[Issue] invalid memory address with sitedossier source

**Describe the bug**
A clear and concise description of what the bug is.

```bash
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x7d0532]

goroutine 79 [running]:
github.com/projectdiscovery/subfinder/v2/pkg/subscraping/sources/sitedossier.(agent).enumerate(0xc00011f240, 0xa208a0, 0xc000060900, 0xc00002a480, 0x38)
        /root/go/pkg/mod/github.com/projectdiscovery/subfinder/v2@v2.4.5/pkg/subscraping/sources/sitedossier/sitedossier.go:40 +0x192
github.com/projectdiscovery/subfinder/v2/pkg/subscraping/sources/sitedossier.(Source).Run.func1(0xc00011f240, 0xa208a0, 0xc000060900, 0x7fffc2cb9701, 0x10)        /root/go/pkg/mod/github.com/projectdiscovery/subfinder/v2@v2.4.5/pkg/subscraping/sources/sitedossier/sitedossier.go:73 +0xc6
created by github.com/projectdiscovery/subfinder/v2/pkg/subscraping/sources/sitedossier.(*Source).Run
        /root/go/pkg/mod/github.com/projectdiscovery/subfinder/v2@v2.4.5/pkg/subscraping/sources/sitedossier/sitedossier.go:72 +0xc5
```

**Subfinder version**
 2.4.5

--- a/.github/workflows/build.yaml
+++ b/.github/workflows/build.yaml
@@ -13,10 +13,10 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v2
       - name: Run golangci-lint
-        uses: golangci/golangci-lint-action@v1
+        uses: golangci/golangci-lint-action@v2
         with:
           # Required: the version of golangci-lint is required and must be specified without patch version: we always use the latest patch version.
-          version: v1.31
+          version: v1.33
           args: --timeout 5m
           working-directory: v2/
 

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -111,6 +111,6 @@ linters:
 # golangci.com configuration
 # https://github.com/golangci/golangci/wiki/Configuration
 service:
-  golangci-lint-version: 1.31.x # use the fixed version to not introduce new linters unexpectedly
+  golangci-lint-version: 1.33.x # use the fixed version to not introduce new linters unexpectedly
   prepare:
     - echo "here I can run custom commands, but no preparation needed for this repo"

--- a/v2/go.mod
+++ b/v2/go.mod
@@ -12,6 +12,7 @@ require (
 	github.com/projectdiscovery/fdmax v0.0.3
 	github.com/projectdiscovery/gologger v1.0.1
 	github.com/rs/xid v1.2.1
+	github.com/stretchr/testify v1.5.1
 	github.com/tomnomnom/linkheader v0.0.0-20180905144013-02ca5825eb80
 	gopkg.in/yaml.v3 v3.0.0-20200615113413-eeeca48fe776
 )

--- a/v2/pkg/subscraping/sources/sitedossier/sitedossier.go
+++ b/v2/pkg/subscraping/sources/sitedossier/sitedossier.go
@@ -35,12 +35,14 @@ func (a *agent) enumerate(ctx context.Context, baseURL string) {
 	if err != nil && !isnotfound {
 		a.results <- subscraping.Result{Source: "sitedossier", Type: subscraping.Error, Error: err}
 		a.session.DiscardHTTPResponse(resp)
+		return
 	}
 
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		a.results <- subscraping.Result{Source: "sitedossier", Type: subscraping.Error, Error: err}
 		resp.Body.Close()
+		return
 	}
 	resp.Body.Close()
 

